### DISCUSSION TITLE
Moving `main.go` to `cmd`

### DISCUSSION THREAD

**Discussion by paganotoni** [gobuffalo/cli#115] [Ideas]
There is a common pattern in the Go community to put the `main` packages within the `cmd` folder, including this repo. However, Buffalo generated applications add the main.go file in the root folder of the codebase.

I wonder if it would be a good idea to move Buffalo generated apps to have their `main.go` file live within the `cmd/[appname]` folder, and in that way follow this common Go pattern.

Implementing this change would mean (From the top of my head):

- Updating the generated application templates
- Updating the build process
- Adding a fixer to move the file

Thoughts? 

cc @sio4 @fasmat @stanislas-m

**Comment by fasmat** [comment]
I'd be ok with the change. Wasn't bothered by the current location of `main.go` but you are right it should be in `cmd` not `cmd/buffalo`

**Comment by sio4** [comment]
That could be a good idea for v1!

For now, there could be some plus and minus:

- It looks more go-ish so gophers will love it.
- clean root.
- By the way, since the buffalo app is not built by `go install` command, the convention could not be important.
- Then cli need to care about both locations or need another fixer. I feel like buffalo cli became more complex and I would like to prevent complexity as much as possible.

Not only for this, IMO, I think it would be better if we start planning the direction of v1 rather than improve/make a change in the current version.  I think collecting random ideas is a good starting point, so I hope we could have a bucket for collecting random ideas for v1. What do you think?

One question. What is your preference for design discussion? I hope to follow/learn your culture or preference. For me, I feel good with a shared doc (like the doc I shared last year).

**Reply by paganotoni** [reply]
Google docs would be good. Then we can have discussions here for each of the items in the roadmap. As a sidenote, that process is something we need to come up with as a team.

**Reply by sio4** [reply]
FYI, the doc I wrote is as follows. Please request access. Not sure if it is better to make it public.

https://docs.google.com/document/d/1yO4xYyTzPVpJeOPbsnbFAUzf1OSSFhN4vOtnSzPlSuM/edit?usp=sharing
